Lib/importlib/util.py

```diff
@@ -116,12 +116,24 @@ def find_spec(name, package=None):
 # is imported by runpy, which means we want to avoid any unnecessary
 # dependencies.  Thus we use a class.
 
-class allowing_all_extensions:
-    """A context manager that lets users skip the compatibility check.
+class _incompatible_extension_module_restrictions:
```

Comment:
```suggestion
class _skip_incompatible_extension_module_restrictions:
```
?

Reply:
It can be used to turn the restrictions back on too.  By default it enforces the restrictions and you have to explicitly tell it to disable the check.

That said, I agree the name may imply that the default is actively changing something normally, which isn't the case.  What if we don't have a default value:

```python
    with importlib.util._incompatible_extension_module_restrictions(disable_check=True):
        import my_module
```

Reply:
ping @brettcannon

Reply:
That makes more sense with the required argument in terms of what's occurring.